### ISSUE TITLE
Resurface of int not working on radio button

### ISSUE THREAD

**Issue by u12206050** [directus/directus#11814]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

As mentioned here int values don't seem to work since the interface is converting it to string
[10267](https://github.com/directus/directus/issues/10267)
[10111](https://github.com/directus/directus/issues/10111)

Saving a radio button that has integer values looks like this in the payload (saving as string)
![Screenshot 2022-02-23 at 09 26 11](https://user-images.githubusercontent.com/6641242/155283770-d8839409-f852-4da8-903a-2b46bda4061f.png)

But the response clearly shows that it is stored and returned as an integer:
![Screenshot 2022-02-23 at 09 27 32](https://user-images.githubusercontent.com/6641242/155283934-374c53dc-64f2-4059-b28e-13f3cf1b5096.png)

Havn't been able to find the cause in the latest commits though

### To Reproduce

Create radio input field on an int column, and only add numbers as values.

### Errors Shown

None, just doesn't show as being selected:
![Screenshot 2022-02-23 at 09 29 06](https://user-images.githubusercontent.com/6641242/155284124-1df3de40-2111-4e14-8649-f4d435b97e8f.png)

### What version of Directus are you using?

9.5.2

### What version of Node.js are you using?

16.4.0

### What database are you using?

mysql 8

### What browser are you using?

chrome

### What operating system are you using?

osx

### How are you deploying Directus?

Locally & GCP

**Comment by azrikahar** [comment]
Unfortunately I can't seem to reproduce this on my end:

### Clip

https://user-images.githubusercontent.com/42867097/155286753-80a9f334-16e9-4d5a-9a31-90aa44940829.mp4

### Schema showing type "Integer"

![image](https://user-images.githubusercontent.com/42867097/155286875-4e59fd18-98c7-4fc4-bd0d-acebc4b24291.png)

### Interface choices

![LU0YSW4uE2](https://user-images.githubusercontent.com/42867097/155287020-c7238249-1f30-43d6-ba11-fcc3b42ff891.gif)

Were you able to spot any difference between our setups here? Or perhaps can you try with a new Directus instance if possible?

**Comment by u12206050** [comment]
Interesting, no I didn't see any difference between mine and yours. But I then suspected that the options were possibly saved different in earlier version, and so I just updated the field by making a change so it will resave the options and now it works!

So looking in the history the values before were actually saved as strings in the field options, where as now they are saved as integer.

Thanks for the detailed response.
